docs/pages/enterprise/hsm.mdx

```diff
@@ -0,0 +1,205 @@
+and signing, include the `ca_key_params` section in `/etc/teleport.yaml` on the
+auth server.
+
+<Tabs>
+  <TabItem label="AWS CloudHSM">
+```yaml
+# /etc/teleport.yaml
+teleport:
+  ...
+
+auth_service:
+  enabled: true
+  ...
+
+  ca_key_params:
+    pkcs11:
+      module_path: /opt/cloudhsm/lib/libcloudhsm_pkcs11.so
+      # token_label should always be "cavium" for CloudHSM
+      token_label: "cavium"
+      pin: "<CU_username>:<CU_password>"
+      # pin_path can optionally be used to read the pin from a file
+      # pin_path: /path/to/pin_file
+```
+  </TabItem>
+
+  <TabItem label="YubiHSM2">
+```yaml
+# /etc/teleport.yaml
+teleport:
+  ...
+
+auth_service:
+  enabled: true
+  ...
+
+  ca_key_params:
+    pkcs11:
+      module_path: /usr/local/lib/pkcs11/yubihsm_pkcs11.dylib
+      # slot_number should be set to 0 for YubiHSM2
+      slot_number: 0
+      # pin should be the (hex) slot of your authentication key,
+      # concatenated with the password
+      pin: "53bfhunter22"
+      # pin_path can optionally be used to read the pin from a file
+      # pin_path: /path/to/pin_file
+```
+  </TabItem>
+</Tabs>
+
+## Step 3/5. (Re)start Teleport Auth
+
+If this is a new auth server which has not been started yet, starting a brand
+new cluster with an empty backend, HSM keys will be automatically generated at
+startup and no further action is required, skip to step 5. Otherwise, continue
+reading.
+
+If you are connecting an HSM to an existing Teleport cluster, restart the auth
+server for the configuration changes to take effect. New CA keys will
+automatically be generated in the HSM. For these keys to be trusted by the rest
+of the cluster you will need to permorm a CA rotation, see
+[Step 4](#step-45-certificate-rotation-with-hsm). The auth server will not perform
+any signing operations until the rotation has started. In an HA cluster you
+should add the HSM to the auth configuration one server at a time, and do not
+route any traffic to the auth server where the HSM is currently being added.
+
+## Step 4/5. Certificate Rotation with HSM
+
+When adding a new HSM to an existing Teleport cluster, or adding a new
+HSM-connected Auth server to an HA Teleport cluster, CA rotation needs to be
+performed in order for the cluster to trust the new HSM certificates. `teleport`
+will print a warning during startup if this needs to be completed, and will not
+sign any certificates (except the `Admin` certificate used by `tctl` which will
+be signed by a temporary HSM key). A warning will also be printed in `tctl
+status` if this is required for any Auth server in the cluster.
+
+CA rotation can be performed manually or semi-automatically, see our admin guide
+on [Certificate rotation](../setup/operations/ca-rotation.mdx). To rotate the
+CAs manually you can run:
+
+```code
+$ tctl status
+WARNING: One or more auth servers has a newly added or removed HSM...
+$ tctl auth rotate --manual --phase init
```

Comment:
I'm finding it hard to parse this block - maybe some vertical whitespace might help here.